sbi/inference/posteriors/sbi_posterior.py

```diff
@@ -44,7 +52,8 @@ def __init__(
             prior: Prior distribution with methods `log_prob` and `sample`.
             x_o: Observation acting as conditioning context. Absent if None.
```

Comment:
I added the following to the docstring:

```
x_o: Observation acting as conditioning context. Has two purposes:
        1) when running the inference across multiple rounds, we sample from the
        posterior $p(theta|x_o)$ to obtain parameters for the next round. Hence,
        the posterior will be more accurate for $x_o$ than for other
        $x \neq x_o$.
        2) the posterior object returned by the inference method will use $x_o$
        as default value. Hence, when calling e.g. posterior.sample(theta),
        it will automatically sample from the posterior given $x_o$, i.e.
        from $p(theta|x_o)$.
        If x_o is None, only single round inference is possible. Also, calling
        `posterior.sample()` and `posterior.log_prob()` after inference always
        requires to explicitly specify some conditioning variable $x$. This
        setting is recommended when doing amortized inference. 
```

Reply:
Now thinking about this: The docstring I just posted should go to `NeuralInference`, not here. Will move it.

Reply:
Not moving it, see #191